packages/ui/react-ui-table/src/model/table-presentation.ts

```diff
@@ -0,0 +1,202 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { signal } from '@preact/signals-core';
+
+import { FormatEnum, getValue } from '@dxos/echo-schema';
+import { cellClassesForFieldType, cellClassesForRowSelection, formatForDisplay } from '@dxos/react-ui-form';
+import {
+  type DxGridPlane,
+  type DxGridPlaneCells,
+  toPlaneCellIndex,
+  type DxGridPlaneRange,
+  type DxGridCellValue,
+} from '@dxos/react-ui-grid';
+import { mx } from '@dxos/react-ui-theme';
+import { VIEW_FIELD_LIMIT, type FieldType } from '@dxos/schema';
+
+import { type BaseTableRow, type TableModel } from './table-model';
+import { tableButtons } from '../util';
+import { tableControls } from '../util/table-controls';
+
+export class TablePresentation<T extends BaseTableRow = { id: string }> {
```

Comment:
contrary to TableControls, this seems like a good abstraction, but please add a one line comment in the `/** */` block.